#21 - include subject for schedule reminder transactional mails

jitendrapurohit opened the pull request:
This change adds a subject display on Mailing Detail Report as per the requirement of https://github.com/fuzionnz/nz.co.fuzion.transactional/issues/21.

@lcdservices Currently, it only supports the display of Schedule Reminder Emails that is recorded as Transactional by this extension. To test this -

- Apply this patch and execute the upgrade on the Extension page. It adds a new table `civicrm_transactional_mapping` to map the group name with the entity that is sent. Eg Action Schedule id.

- Create a Sched Reminder for Membership, Contribution, etc and send it to a contact by executing the "Send Schedule Reminder" job.

- Check the Mailing Report. It should now display the subject of the reminder that was sent to the user.

Let us know if you want this ext. to show the subject for other transactional emails too, eg contribution receipt, membership receipt, etc. 

Thanks.

Commit:
#21 - include subject for schedule reminder transactional mails

Commit:
add subject for activity emails

lcdservices commented:
@jitendrapurohit This did not work for me. I applied the update, sent myself an activity email, ran the report, and though I could see the new email included in the report, there was no subject value.

On a side note -- 
The recent modifications add a mailing queue id column which can't be removed from the display, and is pretty meaningless to end users. I think you've added that column so it can be referenced when altering the rows. If you pass ```'no_display' => TRUE,``` with the params where that is defined, it will suppress it from view, but should still be available to use in the query.

jitendrapurohit commented:
@lcdservices ok - The previous update was for activity assignee emails to display subject in the reports. I've now updated this PR to also show the subject for the direct email sent by a user. 

Can you verify if it is now working as expected? Also removed the queue id from the report columns.

Thanks.

lcdservices commented:
@jitendrapurohit Looks good. Thanks.

jitendrapurohit commented:
cool, merging it based on above comment.
